sdk/keyvault/keyvault-secrets/package.json

```diff
@@ -116,7 +116,7 @@
     "mocha-junit-reporter": "^1.18.0",
     "mocha-multi": "^1.1.3",
     "nise": "^1.4.10",
-    "nock": "^10.0.6",
+    "nock": "^11.3.2",
```

Comment:
@sadasant   
`@azure/test-utils-recorder` is imported by keyvault-secrets, right ? 
Why is it still depending on `nock` ?

Reply:
If we want to change this, please do it in a separate PR (either before or after this PR is merged)

Reply:
I'm pretty sure we don't need nock here. I most certainly just forgot to remove it. Can we remove it as part of this PR?

Reply:
I'd still rather do this in a separate PR.  I want to keep the version update PRs pure.

Reply:
Sounds good! Here's an issue: https://github.com/Azure/azure-sdk-for-js/issues/5003